### PR TITLE
Add doc

### PR DESCRIPTION
**Pull request by bakura10**
This add doc for the new PR.

ping @Ocramius

### CODE REVIEW

File: docs/04. Controllers.md [hunk 1]
```diff
@@ -164,43 +126,132 @@ but the `/users/4/tweets` will be dispatched to the `Tweet\Controller\UserTweetL
...
+
+* `ZfrRest\Mvc\Controller\Event\ValidationEvent::EVENT_VALIDATE_PRE`: this event is triggered before ZfrRest validates
+any data. You can use this event to override the input filter (that is fetched from mapping, by default), to prevent
+auto filtering...
+* `ZfrRest\Mvc\Controller\Event\ValidationEvent::EVENT_VALIDATE_ERROR`: this event is triggered when an input filter
+validation fails.
+* `ZfrRest\Mvc\Controller\Event\ValidationEvent::EVENT_VALIDATE_SUCCESS`: this event is triggered when an input filter
+validation is successful.
+* `ZfrRest\Mvc\Controller\Event\HydrationEvent::EVENT_HYDRATE_PRE`: this event is triggered before ZfrRest automatically
+hydrates an object. You can use this event to prevent this behaviour.
```
Comment (danizord) [inline docs/04. Controllers.md]: Can also use it to override the hydrator

File: docs/04. Controllers.md [hunk 2]
```diff
@@ -164,43 +126,132 @@ but the `/users/4/tweets` will be dispatched to the `Tweet\Controller\UserTweetL
...
+        parent::attachDefaultListeners(); // Don't forget that!
+
+        $eventManager = $this->getEventManager();
+        $eventManager->attach(ValidationEvent::EVENT_VALIDATION_PRE, [$this, 'preventAutoFiltering']);
+        $eventManager->attach(HydrationEvent::EVENT_HYDRATION_PRE, [$this, 'preventAutoHydration']);
+    }
+
+    public function preventAutoFiltering(ValidationEvent $event)
+    {
+        $method = strtolower($this->request->getMethod());
```
Comment (danizord) [inline docs/04. Controllers.md]: You can use `if ($this->request->isPost())` instead.

File: docs/04. Controllers.md [hunk 3]
```diff
@@ -164,43 +126,132 @@ but the `/users/4/tweets` will be dispatched to the `Tweet\Controller\UserTweetL
...
 
+    public function configureInputFilter(ValidationEvent $event)
+    {
         $method = strtolower($this->request->getMethod());
 
-        // We want to keep all the fields for POST, but limit them for PUT
-        if ($method === 'put') {
-            $inputFilter = $inputFilter->setValidationGroup(['last_name', 'password']);
+        // If method is not PUT, we can return early
+        if ($method !== 'put') {
```
Comment (danizord) [inline docs/04. Controllers.md]: `if (!$this->request->isPut())`

Comment (bakura10) [inline docs/04. Controllers.md]: You're right. Maybe the simpler is you to merge and make the changes, as those pr are sent to your local branches.

Envoyé de mon iPhone

Comment (danizord) [inline docs/04. Controllers.md]: Ok

**Comment by bakura10** [comment]
Nevermind, I did it :). I think it can be merged ;)

**Pull request merged**